uniffi/src/support/mod.rs

```diff
@@ -158,3 +158,61 @@ unsafe impl<T: Liftable + Lowerable> ViaFfi for Option<T> {
         try_lift(v)
     }
 }
+
+unsafe impl<'a> ViaFfi for &'a str {
+    type Value = ffi_support::FfiStr<'a>;
+    fn try_from_ffi_value(v: Self::Value) -> Result<Self> {
+        Ok(v.as_str())
+    }
+
+    // This should never happen since there is asymmetry with strings
+    // We typically go FfiStr -> &str for argumetns and
+    // String -> *mut c_char for return values.
+    // We panic for now, the ViaFfi triat will be
+    // broken down to IntoFfi and TryFromFfi to prevent us from having
+    // to add this
+    fn into_ffi_value(self) -> Self::Value {
+        panic!("Invalid conversion. into_ffi_value should not be called on a &str, a String -> *mut c_char conversion should be used instead.")
+    }
+}
+
+unsafe impl ViaFfi for String {
+    type Value = *mut std::os::raw::c_char;
+    fn into_ffi_value(self) -> Self::Value {
+        ffi_support::rust_string_to_c(self)
+    }
+
+    // This should never happen since there is asymmetry with strings
+    // We typically go FfiStr -> &str for argumetns and
+    // String -> *mut c_char for return values.
+    // We panic for now, the ViaFfi triat will be
+    // broken down to IntoFfi and TryFromFfi to prevent us from having
+    // to add this
+    fn try_from_ffi_value(_: Self::Value) -> Result<Self> {
+        panic!("Invalid conversion. try_from_ffi_value should not be called on a c_char, an FfiStr<'_> -> &str conversion should be used instead.")
+    }
+}
+
+impl Lowerable for String {
+    fn lower_into<B: BufMut>(&self, buf: &mut B) {
+        assert!((self.len() as u64) < u32::MAX as u64);
+        buf.put_u32(self.len() as u32); // We limit strings to u32::MAX bytes
+        buf.put(self.as_bytes());
+    }
+}
+
+// Having this be for &str instead of String would have been nice for consistency but...
+// it seems very dangerous and only possible with some unsafe magic
+// and having a slice referencing the buffer seems like a recipie for
+// disaster... so we have a String here instead.
```

Comment:
Ya Hmmmmm. I wish I could say I've used `Cow` much but it kinda makes sense, this could be a nice optimization (if it doesn't add much complexity) for later 👀